src/v/kafka/server/replicated_partition.cc

```diff
@@ -117,7 +118,7 @@ ss::future<storage::translating_reader> replicated_partition::make_reader(
 
 ss::future<std::vector<cluster::rm_stm::tx_range>>
 replicated_partition::aborted_transactions_local(
-  cloud_storage::offset_range offsets,
+  const cloud_storage::offset_range& offsets,
```

Comment:
Pass by reference into coroutine. Is this because it is now a forward declaration?

Reply:
good spot. the only caller i see `co_return co_await aborted_transactions_local(offsets, ot_state);` which suggests this can continue to be pass-by-value (but the call side can updated with std::move).

Reply:
in this case I assumed that the lifetime of the `offsets` will be long enough since it's awaited in coroutine
do you mean that there could be another callers that might use it the wrong way?

Reply:
the lifetime looks ok (the caller awaits like you point out). but what we want to do is take the approach that parameters to co-routines should be by default pass-by-value, reserving pass by reference only for performance critical scenarios. this way we minimize the number of places in the code base where someone would need to examine all call sites to evaluate lifetime semantics of parameters.

Reply:
fixed that